### PR TITLE
Translations

### PR DESCRIPTION
**Pull request by YishaiGlasner**
This PR is for new feature - in the translation box instead of seeing the version title, the main thing will be the translation itself.
It uses API v3 for getting the text.
New functions for calling it are in sefaria.js.
Now VersionBlock, when called by TranslationBox, renders new VersionBlockWithPreview rather than VersionBlock.
Some functions of VersionBlock that in use in both components, moved to a new class (in VersionBlock.jsx).
Also some new components refactored out of VersionBlock: VersionBlockHeader, VersionBlockSelectButton, VersionDetailsInformation, VersionDetailsImage.
Other new components: VersionBlockWithPreviewTitleLine , VersionPreviewMeta.
Another component that have been changed to feet the new component - OpenConnectionTabButton.

### CODE REVIEW

File: static/js/VersionBlockHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          </a>
+    );
+}
+VersionBlockHeaderTitle.prototypes = {
+  onClick: PropTypes.func.isRequired,
+  versionTitle: PropTypes.string.isRequired,
+  link: PropTypes.string.isRequired,
+};
+
+function VersionBlockHeaderText({link, onClick, text, direction}) {
```
[the remark this answers is not in the document]
Comment (edamboritz) [reply static/js/VersionBlockHeader.jsx:42]: Most of what is being done here can be accomplished more easily and cleanly using css (line-clamp and overflow rules). The only thing we may need js for is to create a single state toggle for toggling a collapsed / expanded verse toggle that changes the css class that affects if the css for line clamping is enabled.

**Review by edamboritz — COMMENTED** [review]
This is very good work! 
I think there are 3 main takeaways here that we should consider changing before merging (explained in line in further detail): 
1. The nested function definitions
2. We should group the components in modules differently. Perhaps all ther version block related shared code in one module and move version block list to a separate one
3. The overly complicated handling of line truncation in the header title text.

File: static/js/sefaria/strings.js [hunk 2]
```diff
@@ -243,6 +243,8 @@ const Strings = {
     "Select Version": "בחירת מהדורה",
```
[the remark this answers is not in the document]
Comment (YishaiGlasner) [reply static/js/sefaria/strings.js:243]: we use Select Version and also Select Translation

File: static/js/sefaria/sefaria.js [hunk 3]
```diff
@@ -487,6 +487,35 @@ Sefaria = extend(Sefaria, {
...
+    // requiredVersions is array of objects that can have language and versionTitle
+    function makeParamsString(language, versionTitle) {
+      if (versionTitle) {
+        return `${language}|${versionTitle}`;
+      } else if (language) {
+        return language;
+      }
+    }
+    function makeUrl() {
+      const host = Sefaria.apiHost;
```
[the remark this answers is not in the document]
Comment (YishaiGlasner) [reply static/js/sefaria/sefaria.js:501]: i'm not sure i understood this (and neither Noah)

**Pull request closed without merging**